app/helpers/i18n_helper.rb

```diff
@@ -0,0 +1,17 @@
+module I18nHelper
+
+  private
+  def set_locale
+    # update current_spree_user locale if logged in and set cookie locale if params
+    if params[:locale] && Rails.application.config.i18n.available_locales.include?(params[:locale])
+      current_spree_user.update_attributes!(locale: params[:locale]) if current_spree_user
```

Comment:
I'm sure `current_spree_user` and `spree_current_user` are the same thing (correct me if that is not right). We have been using `spree_current_user` everywhere else in the code, so would be good to stick with that convention.